File: PERL/Cap07/c07ex03.pl
# c07ex03.pl

#!/usr/bin/perl

use strict;
use warnings;

our ($A);
our ($B); 

sub TROCA {
  my $X = $A;
  $A = $B;
  $B = $X;
}

print "Entre um valor para a variavel <A>: ";
$A = <STDIN>;
chomp($A);

print "Entre um valor para a variavel <B>: ";
$B = <STDIN>;
chomp($B);

TROCA();

print "\n";
print "Variavel <A> com valor $A\n";
print "Variavel <B> com valor $B\n";

print "\n";
print "Tecle <Enter> para encerrar... ";
<STDIN>;

File: C/Cap07/c07ex03.c
// c07ex03.c

#include <stdio.h>

int A, B;

void TROCA(void) {
  int X = A;
  A = B;
  B = X;
}

int main(void) {
  printf("Entre um valor para a variavel <A>: ");
  scanf("%d", &A);
  while (getchar() != '\n');

  printf("Entre um valor para a variavel <B>: ");
  scanf("%d", &B);
  while (getchar() != '\n');

  TROCA();

  printf("\n");
  printf("Variavel <A> com valor %d\n", A);
  printf("Variavel <B> com valor %d\n", B);

  printf("\n");
  printf("Tecle <Enter> para encerrar... ");
  while (getchar() != '\n');

  return 0;
}
